src/App.js

```diff
@@ -1,10 +1,122 @@
 import React from 'react';
+import Header from './components/Header';
+import Main from './components/Main';
+import Footer from './components/Footer';
 
-function App() {
-  return (
-    <div className="App">
-    </div>
-  );
-}
+export default class App extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            todoData: [],
+            filter: 'all'
+        };
+
+    }
+    flag = true;
+    counter = 1;
+
+    createTodoItem(text) {
+        return {
+            text,
+            done: false,
+            id: this.counter++
+        }
+    }
 
-export default App;
+    addItem = (newTodo) => {
+        if (newTodo === '') {
+            return;
+        }
+        this.setState(({todoData}) => {
+                let arrTodoData = [...todoData, this.createTodoItem(newTodo)];
+                return {todoData: arrTodoData}
+            }
+        )
+    };
+    findElement = (todoData, id) => {
+        return todoData.findIndex((el) => el.id === id);
+    };
+
+    onChecked = (id) => {
+        this.setState(({todoData}) => {
+            const copy = [...todoData];
+            const index = this.findElement(copy, id);
+            let obj = copy[index];
+            obj.done = !obj.done;
+            return {
+                todoData: copy
+            }
+        })
+    };
+    checkAll = () => {
+        this.setState(({todoData}) => {
+            const elements = [...todoData];
+            elements.forEach(el => {
+                el.done = this.flag;
+            });
+            this.flag = !this.flag;
+            return {
+                todoData: elements,
+            };
+        })
+    };
+    deleteAll = () => {
+        this.setState(({todoData}) => {
+            const arr = todoData.filter(el => !el.done);
+            return {
+                todoData: arr
+            }
+        });
+    };
+    deleteItem = (id) => {
+        this.setState(({todoData}) => {
+            const index = this.findElement(todoData, id);
+            const newArr = [
+                ...todoData.slice(0, index),
+                ...todoData.slice(index + 1)
+            ];
+            return {
+                todoData: newArr
+            }
+        });
+    };
+    showActive = () => {
+        return this.state.todoData.filter(el => !el.done).length;
+    };
+    filter = (items, filter) => {
```

Comment:
```suggestion
    filter = (filterBy) => {
```
`items` is redundant. You filter only todo-list